### PR TITLE
Add test plan example

### PR DESCRIPTION
**Pull request by darrenbiel**
Add an example and README for a test plan using execution actions to execute a test program on a remote system.

### CODE REVIEW

**Comment by darrenbiel** [comment]
It looks like I don't have permission to add reviewers, but this is ready. @TJ-G

**Review by kylesull30 — APPROVED** [review]

File: Test Plan Operations Examples/README.md [hunk 1]
```diff
@@ -0,0 +1,154 @@
+# Test Plan Operations Example
+
+This example demonstrates how to define a test plan template with execution
+actions that can be used to execute a test plan on a target system.
+
+## Test plan templates
+
+Test plan templates appear in the Work Order UI when creating a new test plan
```
Comment (TJ-G) [inline Test Plan Operations Examples/README.md:8]: I would change to "A test plan template can be specified to provide product specific default values when creating a new test plan."

File: Test Plan Operations Examples/README.md [hunk 2]
```diff
@@ -0,0 +1,154 @@
...
+and provide default values for the test plan. Additionally, templates can
+include execution actions that allow for customized action buttons to appear in
+the test plan details UI. These actions are copied to the test plan when
```
Comment (TJ-G) [inline Test Plan Operations Examples/README.md:11]: suggest: "the test plan details page."

**Review by TJ-G — APPROVED** [review]
Looks good, just some minor feedback.

File: Test Plan Operations Examples/README.md [hunk 3]
```diff
@@ -0,0 +1,153 @@
...
+
+The `notebookId` field of the action specifies the ID of the notebook to
+execute. The id can be found in the Scripts UI. Navigate to the Analysis
+Development tab and locate the published notebook. Right-click on the notebook
+and select **Edit**. The notebook ID will be displayed in the Edit Published
+Notebook panel.
+
+![Published notebook ID](/.attachments/published-notebook-id.png)
+
+### Job parameters
```
Comment (kylesull30) [inline Test Plan Operations Examples/README.md:68]: `Job arguments`

File: Test Plan Operations Examples/README.md [hunk 4]
```diff
@@ -0,0 +1,153 @@
...
+
+For notebooks, the `parameters` field expects an array. The `testPlanId` and
+`systemId` properties are always passed as parameters to a notebook and do not
+need to be specified.
+
+### Escaping arguments and parameters
+
+When passing string arguments or parameters to a job or notebook, it is
+important to ensure that they are properly escaped. Arguments and parameters
+must be escaped when passed as json to the Work Order service and escaped again
```
Comment (kylesull30) [inline Test Plan Operations Examples/README.md:110]: `JSON`

File: Test Plan Operations Examples/README.md [hunk 5]
```diff
@@ -0,0 +1,153 @@
...
+to be passed to the job or notebook execution. For example, if passing a path as
+an argument or parameter, the backslashes must be escaped once for json parsing
```
Comment (kylesull30) [inline Test Plan Operations Examples/README.md:112]: `JSON`

File: Test Plan Operations Examples/README.md [hunk 6]
```diff
@@ -0,0 +1,153 @@
...
+by the Work Order service and then again to be passed to job or notebook
+executions: `"C:\\\\\\\\path\\\\to\\\\sequence.seq"`.
+
+Refer to the Work Order API Swagger documentation for more details on the schema
+of the actions.
+
+### Argument and parameter property replacement
+
+The arguments and parameters may use property replacement to insert property
```
Comment (kylesull30) [inline Test Plan Operations Examples/README.md:121]: Is it worth mentioning that non-string types aren't supported?

File: Test Plan Operations Examples/README.md [hunk 7]
```diff
@@ -0,0 +1,153 @@
...
+or `<testPlanId>` can be used.
+
+Custom properties may be referenced as `"<properties.property_name>"`. An
+argument or parameter may contain multiple property replacements, such as this
+example containing the path to a sequence file:
+`".\\\\TestPrograms\\\\<partNumber>\\\\<testProgram>.seq`.
+
+Angle brackets `<` and `>` are used to denote parameters. If the argument or
+parameter itself contains angle brackets, they must be escaped with a backslash
+`\`. Additionally, properties may not contain `<`, `>`, or `\` characters.
```
Comment (kylesull30) [inline Test Plan Operations Examples/README.md:135]: `properties` -> `property names`

File: Test Plan Operations Examples/TestPlanTemplate.json [hunk 8]
```diff
@@ -0,0 +1,79 @@
...
+        },
+        {
+          "type": "MANUAL",
+          "action": "ABORT"
+        },
+        {
+          "type": "NOTEBOOK",
+          "action": "END",
+          "notebookId": "1e9cd47e-23fc-4d71-b47e-e38b7a930e42",
+          "parameters": {
```
Comment (kylesull30) [inline Test Plan Operations Examples/TestPlanTemplate.json:60]: I think the most common use case for notebook parameters is just a list of strings.

**Review by kylesull30 — APPROVED** [review]

**Pull request merged**